Include/cpython/pyatomic_msc.h

```diff
@@ -0,0 +1,860 @@
+}
+
+static inline int
+_Py_atomic_compare_exchange_ssize(Py_ssize_t *address, Py_ssize_t expected, Py_ssize_t value)
+{
+#if SIZEOF_SIZE_T == 8
+    return (__int64)expected == _InterlockedCompareExchange64((volatile __int64*)address, (__int64)value, (__int64)expected);
+#else
+    return (long)expected == _InterlockedCompareExchange((volatile long*)address, (long)value, (long)expected);
+#endif
+}
+
+static inline int
+_Py_atomic_compare_exchange_ptr(void *address, void *expected, void *value)
+{
+    return (void *)expected == _InterlockedCompareExchangePointer((void * volatile *)address, (void *)value, (void *)expected);
+}
+
+static inline int
+_Py_atomic_exchange_int(int *address, int value)
+{
+    return (int)_InterlockedExchange((volatile long*)address, (long)value);
+}
+
+static inline int8_t
+_Py_atomic_exchange_int8(int8_t *address, int8_t value)
+{
+    return (int8_t)_InterlockedExchange8((volatile char*)address, (char)value);
+}
+
+static inline int16_t
+_Py_atomic_exchange_int16(int16_t *address, int16_t value)
+{
+    return (int16_t)_InterlockedExchange16((volatile short*)address, (short)value);
+}
+
+static inline int32_t
+_Py_atomic_exchange_int32(int32_t *address, int32_t value)
+{
+    return (int32_t)_InterlockedExchange((volatile long*)address, (long)value);
+}
+
+static inline int64_t
+_Py_atomic_exchange_int64(int64_t *address, int64_t value)
+{
+#if defined(_M_X64) || defined(_M_ARM64)
+    return (int64_t)_InterlockedExchange64((volatile __int64*)address, (__int64)value);
+#else
+    for (;;) {
+        __int64 old_value = *(volatile __int64*)address;
+        __int64 new_value = (__int64)value;
+        if (old_value == _InterlockedCompareExchange64((volatile __int64*)address, new_value, old_value)) {
+            return old_value;
+        }
+    }
+#endif
+}
+
+static inline intptr_t
+_Py_atomic_exchange_intptr(intptr_t *address, intptr_t value)
+{
+    return (intptr_t)_InterlockedExchangePointer((void * volatile *)address, (void *)value);
+}
+
+static inline unsigned int
+_Py_atomic_exchange_uint(unsigned int *address, unsigned int value)
+{
+    return (unsigned int)_InterlockedExchange((volatile long*)address, (long)value);
+}
+
+static inline uint8_t
+_Py_atomic_exchange_uint8(uint8_t *address, uint8_t value)
+{
+    return (uint8_t)_InterlockedExchange8((volatile char*)address, (char)value);
+}
+
+static inline uint16_t
+_Py_atomic_exchange_uint16(uint16_t *address, uint16_t value)
+{
+    return (uint16_t)_InterlockedExchange16((volatile short*)address, (short)value);
+}
+
+static inline uint32_t
+_Py_atomic_exchange_uint32(uint32_t *address, uint32_t value)
+{
+    return (uint32_t)_InterlockedExchange((volatile long*)address, (long)value);
+}
+
+static inline uint64_t
+_Py_atomic_exchange_uint64(uint64_t *address, uint64_t value)
+{
+    return (uint64_t)_Py_atomic_exchange_int64((volatile __int64*)address, (__int64)value);
```

Comment:
same compiler warning ``different 'volatile' qualifiers``, remove volatile? also cast to int64_t, not __int64.